### ISSUE TITLE
Change how global shortcuts work and make canvas shortcuts global

### ISSUE THREAD

**Issue by dvoytenko** [GoogleForCreators/web-stories-wp#2417]
## Feature Description

Dependent bugs: #1781, #1070, #1808.

The canvas is the central part of the editor. As such most of keyboard shortcuts have to apply to the canvas, unless they conflict with another component currently focused.

The goal is to make canvas shortcuts global, but to change the rules to resolve conflicts.

## Alternatives Considered

We tried to make canvas area always focused. But this kind of focus trapping is an anathema in a Web application and breaks a lot of accessibility features.

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria / QA Instructions

1. Add a new text element. Click "Delete". The element should be deleted.
2. Add a new text element. Click "left" or "right". The element should move.
1. Select an element on the canvas, click on the "Text" tab in the library. Click delete. The element should be deleted.
2. Select a text element on the canvas, click on "B" (bold) button in the design panel. Click delete. The element should be deleted.
3. Add a new page. Add an element. Click on the current page in the carousel. Click "left" arrow key. The first page should be selected. The element should't be moved. Click "delete". The page should be remove.
4. Select a text element. Click on a color selector. Observer color picker is open. Click "delete". The element should not be deleted.

## Implementation Brief

1. Make focus trapping should take into account single vs multi-selection. If a single element is selected, this elements frame should be focused. Otherwise, the canvas itself should be focused.
2. All of the canvas key handlers will be moved to global handlers.
3. In most of places where only up/down or left/right were handled, the relevant components should handle both pairs and just no-op them if no implementation is possible. This is done so that key handling is not fragmented: e.g. up/down works and left/right does something else.
4. Global shortcuts are to be enabled on most buttons/checkboxes/etc. I.e. clicking Delete on a button is should trigger a global shortcut.
5. The global shortcuts should be separated by the dialog boundaries. The idea is that shortcuts should not make changes that the user cannot even see, b/c a dialog is blocking their view.

**Comment by csossi** [comment]
Verified in QA
